src/common/k8s-api/__tests__/kube-api.test.ts

```diff
@@ -12,25 +12,21 @@ import { PassThrough } from "stream";
 import type { ApiManager } from "../api-manager";
 import { apiManager } from "../api-manager";
 import type { FetchMock } from "jest-fetch-mock/types";
+import { DeploymentApi, Ingress, IngressApi, Pod, PodApi } from "../endpoints";
+import { getDiForUnitTesting } from "../../../main/getDiForUnitTesting";
 
 jest.mock("../api-manager");
 
 const mockApiManager = apiManager as jest.Mocked<ApiManager>;
 const mockFetch = fetch as FetchMock;
 
-class TestKubeObject extends KubeObject {
-  static kind = "Pod";
-  static namespaced = true;
-  static apiBase = "/api/v1/pods";
-}
```

Comment:
Use of volatile production god-class instead of dedicated and stable Test-class makes unit test brittle.

Reply:
The "actual ones" need to be tested in their own behaviours :)

For the unit tests to be wrong, surely there is something wrong with the design, and not just the tests. Provided they were not just bad.

Reply:
Not optimal, but resolved.